test/unlimited_contract_size.js

```diff
@@ -1,92 +1,70 @@
-    var provider = Ganache.provider({
-      mnemonic,
-      allowUnlimitedContractSize: false,
-      gasLimit: 20000000
-    });
-    var web3 = new Web3(provider);
-    var accounts;
+    let context;
 
-    before("get accounts", function(done) {
-      web3.eth.getAccounts(function(err, accs) {
-        if (err) {
-          return done(err);
-        }
-        accounts = accs;
-        done();
-      });
+    before("Setup provider to disallow unlimited contract size", async function() {
+      const ganacheOptions = {
+        seed,
+        allowUnlimitedContractSize: false,
+        gasLimit: 20000000
+      };
+      context = await initializeTestProvider(ganacheOptions);
     });
 
-    it("should fail deployment", function(done) {
-      let DummyContract = new web3.eth.Contract(contract.abi);
-      DummyContract.deploy({
-        data: contract.bytecode
-      })
-        .send({ from: accounts[0], gas: 20000000 })
-        .then(function(instance) {
-          done(new Error("succeeded deployment when it should have failed"));
-        })
-        .catch(function(_) {
-          done();
-        });
+    it("should fail deployment", async function() {
+      this.timeout(10000);
+      const { web3 } = context;
+      const { abi, bytecode } = contract;
+      const errorMessage = "succeeded deployment when it should have failed";
+      try {
+        await deploy(abi, bytecode, web3, { gas: 20000000 });
+        assert.fail(errorMessage);
+      } catch (error) {
+        assert.notStrictEqual(error.message, errorMessage);
+      }
     });
   });
 
   describe("Allow Unlimited Contract Size", function() {
-    var provider = Ganache.provider({
-      mnemonic,
-      allowUnlimitedContractSize: true,
-      gasLimit: 20000000
-    });
-    var web3 = new Web3(provider);
-    var accounts;
+    let context;
+
+    before("Setup provider to allow unlimited contract size", async function() {
+      const ganacheOptions = {
+        seed,
+        allowUnlimitedContractSize: true,
+        gasLimit: 20000000
+      };
 
-    before("get accounts", function(done) {
-      web3.eth.getAccounts(function(err, accs) {
-        if (err) {
-          return done(err);
-        }
-        accounts = accs;
-        done();
-      });
+      context = await initializeTestProvider(ganacheOptions);
     });
 
-    it("should succeed deployment", function(done) {
-      let DummyContract = new web3.eth.Contract(contract.abi);
-      DummyContract.deploy({
-        data: contract.bytecode
-      })
-        .send({ from: accounts[0], gas: 20000000 })
-        .then(function(instance) {
-          done();
-        })
-        .catch(function(_) {
-          done(new Error("failed deployment when it should have succeeded"));
-        });
+    it("should succeed deployment", async function() {
+      const { web3 } = context;
+      const { abi, bytecode } = contract;
+      await deploy(abi, bytecode, web3, { gas: 20000000 });
```

Comment:
This should use `await assert.doesNotReject(deploy(abi, bytecode, web3, { gas: 20000000 }), "failed deployment when it should have succeeded");` (https://nodejs.org/docs/latest-v8.x/api/assert.html#assert_assert_doesnotreject_block_error_message) if possible.

Reply:
Can attempt pattern with v8.